File: Week02/2.1P/Counter/Counter/Counter.cs
using System;

namespace Counter
{
    public class Counter
    {
        private int _count;
        private string _name;

        public Counter(string name)
        {
            _name = name;
            _count = 0;
        }

        public void Increase()
        {
            _count += 1;
            return;
        }

        public void Reset()
        {
            _count = 0;
            return;
        }

        public string Name
        {
            get
            {
                return _name;
            }
            set
            {
                _name = value;
            }
        }

        public int Tick
        {
            get
            {
                return _count;
            }
        }
    }
}

File: Week11/11.1P/counter.py
from itertools import count


class Counter:
    def __init__(self, name):
        self.name = name
        self.count = 0

    def Increment(self):
        self.count += 1
    
    def Reset(self):
        self.count = 0

    def Name(self):
        return self.name

    def Tick(self):
        return self.count
    
    

